content/docs/user-guide/checkpoints.md

```diff
@@ -0,0 +1,425 @@
+
+You can add experiments to your Git history by committing the experiment you
+want to track, which you'll see later in this tutorial.
+
+## Setting up the project
+
+This tutorial is going to cover how to implement checkpoints in a ML project
+using DVC. We're going to train a model to identify handwritten digits based on
+the MNIST dataset.
+
+You can follow along with the steps here or you can clone the repo directly from
+GitHub and play with it. To clone the repo, run the following commands.
+
+```bash
+git clone https://github.com/iterative/checkpoints-tutorial
+cd checkpoints-tutorial
+```
+
+It is highly recommended you create a virtual environment for this example. You
+can do that by running: `python3 -m venv .venv`.
+
+Once your virtual environment is installed, you can start it up with one of the
+following commands.
+
+- On Mac/Linux: `source .venv/bin/activate`
+- On Windows: `.\.venv\Scripts\activate`
+
+Once you have your environment set up, you can install the dependencies by
+running: `pip install -r requirements.txt`. This will download all of the
+packages you need to run the example. Now you have everything you need to get
+started with experiments and checkpoints.
+
+## Enabling DVC pipelines for checkpoint experiments
+
+In the `dvc.yaml` file, we need to add a few things to get our checkpoints and
+metrics tracking setup. First we'll add metrics tracking to our experiments. At
+the end of the file, add the following lines.
+
+```yaml
+live:
+  dvclive:
+    summary: true
+    html: true
+```
+
+Your `dvc.yaml` should look similar to this now.
+
+```yaml
+---
+plots:
+  - predictions.json:
+      cache: false
+      template: confusion
+      x: actual
+      y: predicted
+live:
+  dvclive:
+    summary: true
+    html: true
+```
+
+Now we'll set up checkpoints. Below the `model.pt`, add the following code:
+`checkpoint: true`. This enables checkpoints at the pipeline level. Your
+`dvc.yaml` should look like this now.
+
+```yaml
+stages:
+  download:
+    cmd: python download.py
+    deps:
+      - download.py
+    outs:
+      - data/MNIST
+  train:
+    cmd: python train.py
+    deps:
+      - data/MNIST
+      - train.py
+    params:
+      - seed
+      - lr
+      - weight_decay
+    outs:
+      - model.pt:
+          checkpoint: true
+    plots:
+      - predictions.json:
+          cache: false
+          template: confusion
+          x: actual
+          y: predicted
+    live:
+      dvclive:
+        summary: true
+        html: true
+```
+
+The checkpoints need to be enabled in DVC at the pipeline level. The
+`--checkpoint` option of the `dvc stage add` command defines the checkpoint file
+or directory. This will create a new stage with checkpoints enabled, but does
+not update existing stages.
+
+If you want to update an existing stage, adding the `-f` option to the
+`dvc stage add --checkpoint` command will force an overwrite of the current
+stage and update it with checkpoints enabled.
```

Comment:
Do we need to explain `-f` option here? 😄  If the pipeline was defined outside of this it might be better to to jump in the description of `dvc stage add`.

An alternative approach - remove DVC pipeline from the repo and do `dvc init & dvc stage add ...` as a part fo the tutorial. It will reduce the amount of code here and the amount of manual work - copy-pasting a command should be easier than edit dvc.yaml. I like this approach slightly more.